compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -626,6 +626,17 @@ where
         self.write_immediate(Immediate::Scalar(val.into()), dest)
     }
 
+    /// Write a scalar pair to a place
+    #[inline(always)]
+    pub fn write_scalar_pair(
```

Comment:
I don't think we want this helper. We have immediates specifically to represent these pairs. Also this encourages to have scalar pairs without having their type, which is dangerous. I think we actually want to change `write_immediate` to take an `ImmTy` instead of an `Immediate`, but that's a larger change... but this helper moves us in the wrong direction IMO.

Reply:
I like the idea of having typed immediates everywhere, I just know that this isn't what the code is doing right now, and that's why I added this method, since it's either this or import `Immediate` directly into the intrinsics module and construct one myself.

Reply:
> or import Immediate directly into the intrinsics module and construct one myself.

Yes please do that.

Reply:
I had assumed that the presence of `write_scalar` was to avoid that, but I'll keep that in mind. Perhaps `write_scalar` should also be removed if the goal is to have typed immediates everywhere?

Reply:
Scalars are much less at risk of odd effects due to bad types since there's no field offsets / padding being computed.